### PR TITLE
fix(TopSites): Keep placeholder edit button from opening two modals, …

### PR DESCRIPTION
**Pull request by dmose**
…Bug 1431078.  Basically just clones the behavior from non-placeholder cards.

r?@Mardak

### CODE REVIEW

File: system-addon/content-src/components/TopSites/TopSite.jsx [hunk 1]
```diff
@@ -241,10 +241,7 @@ export class TopSitePlaceholder extends React.PureComponent {
     this.onEditButtonClick = this.onEditButtonClick.bind(this);
   }
   onEditButtonClick() {
-    this.props.dispatch({
-      type: at.TOP_SITES_EDIT,
-      data: {index: this.props.index}
-    });
+    this.props.onEdit(this.props.index);
```
Comment (Mardak) [inline system-addon/content-src/components/TopSites/TopSite.jsx]: Probably just make it conditionally call `this.props.onEdit` or do the existing `dispatch` ?

Comment (Mardak) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: @rlr ^^ ?

Comment (rlr) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: yeah, right now clicking on the edit icon from outside the modal doesn't work. I wonder if there is a way we could just kill the whole edit modal concept. I'll discuss with @bryanbell next week.

Comment (dmose) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: @Mardak @rlr: I'm confused.  If I'm understanding what you guys have said correctly, something should be busted with the patch as written.  In my local tree, clicking the hovering edit icon brings up a functional edit modal both when done in the normal topsites or in the edit mode modal.   Am I missing something?

Comment (dmose) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: The hovering edit icon over the giant image for a given site, that is.

Comment (rlr) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: This is what I see (and in the console I see: `JavaScript error: resource://activity-stream/data/content/activity-stream.bundle.js, line 2887: TypeError: this.props.onEdit is not a function`):

Comment (dmose) [reply system-addon/content-src/components/TopSites/TopSite.jsx]: @rlr thanks!   I was clicking the wrong thing.

**Comment by as-pine-proxy** [comment]
[Treeherder: 712df8f61d71c211cb37e97df328fb88d33cdfbb](https://treeherder.mozilla.org/#/jobs?repo=pine&revision=712df8f61d71c211cb37e97df328fb88d33cdfbb)

**Pull request merged**